packages/nimble-components/src/table-column/icon/tests/table-column-icon.spec.ts

```diff
@@ -0,0 +1,396 @@
+import { html, repeat, ref } from '@microsoft/fast-element';
+import { Table, tableTag } from '../../../table';
+import { TableColumnIcon, tableColumnIconTag } from '..';
+import { waitForUpdatesAsync } from '../../../testing/async-helpers';
+import { type Fixture, fixture } from '../../../utilities/tests/fixture';
+import type { TableRecord } from '../../../table/types';
+import { TablePageObject } from '../../../table/testing/table.pageobject';
+import { wackyStrings } from '../../../utilities/tests/wacky-strings';
+import { getSpecTypeByNamedList } from '../../../utilities/tests/parameterized';
+import { mappingTextTag } from '../../../mapping/text';
+import { MappingIcon, mappingIconTag } from '../../../mapping/icon';
+import { IconXmark, iconXmarkTag } from '../../../icons/xmark';
+import { IconCheck, iconCheckTag } from '../../../icons/check';
+import type { MappingKey } from '../../../mapping/base/types';
+
+interface SimpleTableRecord extends TableRecord {
+    field1?: MappingKey | undefined;
+    field2?: MappingKey | undefined;
+}
+
+interface BasicIconMapping {
+    key?: MappingKey;
+    text?: string;
+    icon: string;
+}
+
+class Model {
+    public col1!: TableColumnIcon;
+}
+interface ModelFixture<T> extends Fixture<T> {
+    model: Model;
+}
+
+describe('TableColumnIcon', () => {
```

Comment:
This should also include the _should export its tag_ and _can construct an element instance_ tests.

Reply:
None of our column types had these tests, so I added them.